### PR TITLE
[dtensor][be] Reduced redundant LOC by creating functions to set up models used in example

### PR DESCRIPTION
**Pull request by sinhaanshul**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #130072
* #129994
* __->__ #129613
* #129602

**Summary**
As the CommModeFeature example file grew, there were to many LOC that was repeated for setting up the models used. I created two functions, one to handle MLP and MLPStacked models and the other for transformer models. The output of the examples will not have changed.

**Test Plan**
1. torchrun --standalone --nnodes=1 --nproc-per-node=4 torch/distributed/_tensor/examples/comm_mode_features_example.py -e MLP_distributed_sharding_display

2. torchrun --standalone --nnodes=1 --nproc-per-node=4 torch/distributed/_tensor/examples/comm_mode_features_example.py -e MLPStacked_distributed_sharding_display

3. torchrun --standalone --nnodes=1 --nproc-per-node=4 torch/distributed/_tensor/examples/comm_mode_features_example.py -e MLP_module_tracing

4. torchrun --standalone --nnodes=1 --nproc-per-node=4 torch/distributed/_tensor/examples/comm_mode_features_example.py -e transformer_module_tracing

5. torchrun --standalone --nnodes=1 --nproc-per-node=4 torch/distributed/_tensor/examples/comm_mode_features_example.py -e MLP_operation_tracing

6. torchrun --standalone --nnodes=1 --nproc-per-node=4 torch/distributed/_tensor/examples/comm_mode_features_example.py -e transformer_operation_tracing

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

File: torch/distributed/_tensor/examples/comm_mode_features_example.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# mypy: allow-untyped-defs
```
Comment (tianyu-l) [inline torch/distributed/_tensor/examples/comm_mode_features_example.py:1]: we shouldn't do this unless absolutely necessary?
If it's for `MLPModule` and `MLPStacked`, you can use common ancestor `nn.Module` to denote.

**Review by tianyu-l — COMMENTED** [review]
lgtm other than the typing annotation part. Curious why we need to cancel typing annotation.

File: torch/distributed/_tensor/examples/comm_mode_features_example.py [hunk 2]
```diff
@@ -45,35 +47,62 @@ def __init__(self, world_size: int, rank: int) -> None:
         self.world_size = world_size
         self.rank = rank
         self.device_type = get_device_type()
+        torch.manual_seed(0)
```
Comment (yifuwang) [inline torch/distributed/_tensor/examples/comm_mode_features_example.py:50]: It's probably not a good idea to hide `torch.manual_seed()` inside a `__init__` function.

Also (not related to your change) it's generally a good practice to incorporate rank into the manual seed. Otherwise sharded tensors may be actually be "replicated" numeric-wise. This can hide some issues.

**Review by XilunWu — APPROVED** [review]
LGTM! Thx for the code refacroting!
